### DISCUSSION TITLE
Syntax for running another method from inside method (if possible)

### DISCUSSION THREAD

**Discussion by Youngwhisk** [RomanKubiak/ctrlr#224] [General]
I would like to run (or transfer to) another method from inside a method. I have seen "scriptNameToRun()" mentioned but it doesn't seem to work. I can't find any other references to doing this. In Visual Basic this is simple. I have approx 30 lines of code that needs to be performed each time one of 18 radio buttons is clicked. I think it would be foolish to have to put all that code in each mouse down method.

**Comment by Ted-juh** [comment]
If you write one script that works for all MouseDown methods, it's still one script, right? You can see in the Utility tab that it's just a pointer to that method. It's not copying that script to every radio button. 

You can get values/ results from another function by passing values and return a result. Is that what you want to do?

And have you seen this?
https://ctrlr.org/radio-button-code-generator/

**Reply by Youngwhisk** [reply]
Getting and setting the values is no problem. What I am then doing is setting different text on six different uiLabels based on 2 different modes and 16 different possible channels. That is just for visual information. All the information gathering is fine. The radio buttons need only 2 lines of code for the information to set 2 variables with different values depending on which button is clicked. Actually placing the needed text on the six uiLabels takes approx 14 lines of code which is always the same but generates different text due to the variables. I want to put that 14 lines in its own method and call it at the end of any of the mouse down procedures. I know I can just paste the code into all of the mouse down methods (it does work) but that is inelegant and tedious to change if needed. All I am looking for is a line of code to continue code execution in the other method. I have done that a lot in Visual Basic and I am pretty sure it can be done in Ctrlr/Lua. I just can't find it.

**Reply by dnaldoog** [reply]
Can you post the code you've got so far?

**Reply by Youngwhisk** [reply]
solved, see below

**Comment by Godlike-Productions** [comment]
Just write a method, and call it from within the other method.

eg.

myFirstMethod
Do some stuff
end

mySecondMethod
Do some stuff
myFirstMethod()
Do some more stuff
end

**Reply by Youngwhisk** [reply]
I tried that and it didn't seem to work.  I will try it again.

**Reply by Youngwhisk** [reply]
I had an empty line at the top of the 2nd method. It worked properly after correcting that. However since then something got corrupted and 3/4 of my buttons disappeared from the panel. Something to explore tomorrow.

**Reply by Godlike-Productions** [reply]
Use the modulator list to get them to appear on the panel.  They will get moved to the top left corner when you do this. 

This normally occurs if you have a uiGroup somewhere and try to move things into or out of it, or move things between layers without first taking something out of the uiGroup.

**Reply by Youngwhisk** [reply]
Thanks, I will give that a try.

**Reply by Youngwhisk** [reply]
All the buttons were where you said. Interestingly, one of the Radio Button Groups retained the Radio Button Group number, but the Group Name I had given the buttons and container had reverted back to the name of the modulator container where I had placed them. It was a four word Group Name including spaces. I rewrote the group name for all of the items in that group, shortening it and removing the spaces. Everything is working properly now. I don't know if there are restrictions or limitations on Group Names that caused the problem with my buttons but it all works now.

**Comment by dobo365** [comment]
To be honest, I don't really understand the problem... but maybe I can add a few things here based on my own findings.
I'm re-using all the time my methods for several components so all my panels contains some minimum set of methods.

At the beginning of the code you can track for the name of the component and then set some stuff specific (like the parameter name and some explanation) but then finish the method with something generic for all components of the same nature (displaying the name and the explanation in some labels).

I have also coded in VB and other language but here in Lua and Ctrlr, it is not working the same way and it took me some time to realize it as I was very used to the single flow of execution.
With Ctrlr , when you launch method2 from method1, method1 continues to be executed in parallel to method2. So you must care to not use things set by method2 in that continuation of method1 because method2 may not have reached the needed point when your method1 will need it.
In VB you go from Sub1 to Sub2 then when Sub2 is finished you go back to Sub1 to continue.
It is the reason you need to handle things in a special order, use timers... to be sure things are done correctly.

**Reply by Youngwhisk** [reply]
I believe my problem was leaving a blank line at the top of my method2 code. Removing that fixed the problem. My call to method2 is the last line of code in method1. Everything works correctly now.

Thanks for the information on the 2 methods running in parallel. I would never have thought of that.
